crates/api_models/src/payment_methods.rs

```diff
@@ -820,6 +820,7 @@ pub struct PaymentMethodResponse {
 pub enum PaymentMethodsData {
     Card(CardDetailsPaymentMethod),
     BankDetails(PaymentMethodDataBankCreds),
+    WalletDetails(payments::additional_info::WalletAdditionalDataForCard),
```

Comment:
Can we have a new type here for storage purposes instead of using the payments type? 
We can perform the required transformations wherever necessary.